### PR TITLE
cleanup unused and incorrect annotations

### PR DESCRIPTION
**Pull request by rshriram**

### CODE REVIEW

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -72,9 +46,11 @@ annotations:
...
     resources:
       - Service
-  - name: security.istio.io/autoMTLS
-    description: Determines whether the client proxy uses auto mTLS. This overrides the mesh
-      default specified in MeshConfig.enable_auto_mtls.
+  - name: security.istio.io/tlsMode
+    description: Specifies the TLS mode supported by a sidecar proxy. Valid values are 'istio', 'disabled'.
+      When injecting sidecars into Pods, the sidecar injector will set the value of this label to 'istio' indicating
+      that the sidecar is capable of supporting mTLS. Clients will opportunistically use this label to determine whether
+      or not to secure the the traffic to this workload using Istio mutual TLS.
```
Comment (smawson) [inline annotation/annotations.yaml]:
```suggestion
      or not to secure the traffic to this workload using Istio mutual TLS.
```

File: annotation/annotations.yaml [hunk 2]
```diff
@@ -72,9 +46,11 @@ annotations:
...
     resources:
       - Service
-  - name: security.istio.io/autoMTLS
-    description: Determines whether the client proxy uses auto mTLS. This overrides the mesh
-      default specified in MeshConfig.enable_auto_mtls.
+  - name: security.istio.io/tlsMode
+    description: Specifies the TLS mode supported by a sidecar proxy. Valid values are 'istio', 'disabled'.
+      When injecting sidecars into Pods, the sidecar injector will set the value of this label to 'istio' indicating
+      that the sidecar is capable of supporting mTLS. Clients will opportunistically use this label to determine whether
+      or not to secure the traffic to this workload using Istio mutual TLS.
```
Comment (linsun) [inline annotation/annotations.yaml]: @rshriram are we removing `security.istio.io/autoMTLS` annotation because auto MTLS is globally enabled?

Comment (linsun) [reply annotation/annotations.yaml]: is security.istio.io/tlsMode: istio the same as autoMTLS before?

Comment (hzxuzhonghu) [reply annotation/annotations.yaml]: It was actually replaced

**Pull request merged**